lib/commons/dom/url-props-from-attribute.js

```diff
@@ -0,0 +1,154 @@
+/* global dom */
+
+/**
+ * Parse resource object for a given node from a specified attribute
+ * @method urlPropsFromAttribute
+ * @param {HTMLElement} node given node
+ * @param {String} attribute attribute of the node from which resource should be parsed
+ * @returns {Object}
+ */
+dom.urlPropsFromAttribute = function urlPropsFromAttribute(node, attribute) {
+	const value = node[attribute];
+	if (!value) {
+		return undefined;
+	}
+
+	const nodeName = node.nodeName.toUpperCase();
+	let parser = node;
+
+	/**
+	 * Note:
+	 * The need to create a parser, is to keep this function generic, to be able to parse resource from element like `iframe` with `src` attribute
+	 */
+	if (!['A', 'AREA'].includes(nodeName)) {
+		parser = document.createElement('a');
+		parser.href = value;
+	}
+
+	const { pathname = '', filename = '' } = getPathnameOrFilename(
+		parser.pathname
+	);
+	return {
+		protocol: parser.protocol,
```

Comment:
I'm not so sure we should include this. `http://deque.com` is the same URL as `https://deque.com` I'd say.

Reply:
Upon further reading, the protocol section of the URI is usually called a scheme & I found this long list of schemes - https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml that can form the part of the URI.

At this juncture I am not sure if keeping a finite list of protocols and tweaking it is a good idea (as in convert `https` to `http`). Please provide your thoughts.

Reply:
Even so, "http" and "https" should be treated as the same, same for ftp and ftps. Please add that.